Doc/c-api/module.rst

```diff
@@ -485,10 +485,19 @@ since multiple such modules can be created from a single definition.
 
    Only effective on modules created using single-phase initialization.
 
+   Python calls ``PyState_AddModule`` automatically after importing a module,
```

Comment:
An alternate implementation might not necessarily call ``PyState_AddModule`` but still achieve the same end result. Perhaps just say that Python attaches the module to the interpreter state and should be accessible through ``PyState_FindModule``. How they achieve that should be up to the implementation

Reply:
All ways I can find of wording this more precisely make it significantly harder to understand.

I think it's understood that alternate implementations are free to do an equivalent of calling `PyState_AddModule`, instead of actually calling it. They should still implement the `PyState_AddModule` semantics (unless they decide to differ, of course).

Another thing that's implied here is that Python only calls this on single-phase-init modules. And another is that it's idempotent.

Should we document all that? How much hand-holding do authors of alternative import mechanisms need? (This is a real question, which you are in a good position to answer.)

Reply:
But my main issue is:

> All ways I can find of wording this more precisely make it significantly harder to understand.

If you can think of a better way, let me know! Otherwise I can merge this.